collaboration/extending-kyma-cli.md

```diff
@@ -0,0 +1,230 @@
+
+  ```bash
+  <cliName> <functionality> <action>
+  #example aws s3 ls
+  ```
+
+  The current kyma-cli also uses the same pattern eg.
+
+  ```bash
+  kyma provision
+  or
+  kyma install
+  ```
+
+### Lambdas
+
+For the serverless component we should start with supporting following commands
+
+#### Local workspace setup
+
+1. Creation of project locally
+  `kyma lambda init <lambda-name> --runtime <runtime-name> -n <namespace>` or `kyma lambda init <lambda-name> --runtime <runtime-name> -n <namespace> -p <path>`
+  This should create directoy with following content based on the runtime:
+
+      ```bash
+        <lambda-name>
+        ├── .vscode
+        │   └── launch.json
+        ├── deployment
+        │   └── deployment.yaml
+        ├── local
+        │   └── index.js
+        └── code
+          ├── config.yaml
+          ├── handler.js
+          └── package.json
+      ```
+
+    > When path `-p` is passed then it should create in desitnation path else it use the workspace set in the yaml file.
+
+    * launch.json consists of settings for the debugging
+    * deployment.yaml consists of the yaml that has been deployed on the kyma cluster.
+    * `index.js` consists of the server code with handler pointing to `handler.js`
+    * config.yaml can be used as info file with current configurations like:
+      - name
+      - namespace
+      - events subscribed
+      - environment variables
+      - Api name to expose with
+      - replicas
+      - min/max memory
+      - min/max cpu
+      - runtime
+      - debug command (to be used with telepresence. This can help if we want to have different runtimes)
+    * handler.js where the developer can write his logic to test
+    * package.json file with dependencies.
+
+    >The templates generated should be placed a directory in the kyma cli. In the future if we are supporting more runtimes we can move it to separate repo.
+
+2. Testing/debugging it locally
+
+We can introduce commands to test the lambda locally and also if the developer wants to debug the code. This is useful in case when we have some event payloads that are not easy to recreate.
+
+`kyma lambda debug <lambda-name>`
+
+> We can use telepresence for local debugging. We can either use --swap-deployment or --new-deployment option
+
+#### Working with deployed kyma cluster
+
+We should allow deployment of the lambdas on a k8s cluster. For this we should use the service account role. Following commands should be supported
+
+##### CRUD
+
+`kyma lambda create <lambda-name> -n <namespace> --runtime <runtime> --min-mem <mb> --max-mem <mb> --min-cpu <cpu> --max-cpu <cpu> --min-replicas 1 --max-replicas 1 --src-code <path-to-src-code> --dependecies <path-to-dep-file>`
+
+`kyma lambda create -p <path-to-lambda-dir>`
+> This would create a lambda after reading the deployment.yaml from the path provided and also would read `config.yaml` to read the config like events, api etc.
```

Comment:
Why deployment.yaml, shouldn't be the config.yaml enough?

Reply:
fixed.